tests/feed/test_feed.py

```diff
@@ -0,0 +1,176 @@
+    return post
+
+
+@pytest.fixture
+def persist_post_for_second_course_student(persist_second_course, persist_second_user):
+    post = Post(course_id=persist_second_course,
+                user_id=persist_second_user,
+                parent_post_id=None,
+                content="this is a root post by a second student for second course")
+    post.save()
+    return post
+
+
+@pytest.mark.django_db
+class TestPost:
+    def test_is_root_post(self, persist_post_for_first_course, persist_reply_for_first_course):
+        assert persist_post_for_first_course.is_root_post()
+        assert not persist_reply_for_first_course.is_root_post()
+
+    def test_post_is_only_reply_to_root_post(self, persist_user, persist_course,
+                                             persist_reply_for_first_course):
+        reply_to_reply = Post(course_id=persist_course, user_id=persist_user,
+                              parent_post_id=persist_reply_for_first_course, content="should be invalid")
+        with pytest.raises(ValidationError):
+            reply_to_reply.save()
+
+    def test_is_user_able_to_post_in_course(self, persist_user, persist_course):
+        assert not Post.is_user_able_to_post_in_course(persist_user, persist_course)
+        user_joined_group = StudentCourses(student_id=persist_user, teacher_course_id=persist_course,
+                                           status="Confirmed")
+        user_joined_group.save()
+        assert Post.is_user_able_to_post_in_course(persist_user, persist_course)
+
+    def create_post_for_user_not_related_to_course(self, persist_user, persist_course):
+        assert not Post.is_user_able_to_post_in_course(persist_user, persist_course)
+        invalid_post = Post(course_id=persist_course, user_id=persist_user, content="should be invalid")
+        with pytest.raises(ValidationError):
+            invalid_post.save()
+
+    def test_image_upload(self, image_file, persist_post_for_first_course):
+        persist_post_for_first_course.image = image_file
+        persist_post_for_first_course.save()
+        media_path = os.path.join(settings.MEDIA_ROOT, str(persist_post_for_first_course.image))
+
+        assert persist_post_for_first_course.image.read() == b"file_content"
+        assert os.path.isfile(media_path)
+
+
+@pytest.mark.django_db
+class TestPostManager:
+    def test_get_post_for_user(self, persist_user,
+                               persist_first_student_course, persist_second_student_course,
+                               persist_post_for_first_course, persist_reply_for_first_course,
+                               persist_post_for_second_course_student):
+        expected = {persist_post_for_first_course: [persist_reply_for_first_course]}
+        posts_user = PostManager.get_posts_for_user(persist_user)
+
+        assert posts_user == expected
+        assert persist_post_for_second_course_student not in expected.items()
+
+    @pytest.mark.parametrize("num_of_posts", [4])
```

Comment:
Do we need a parameterized test for only one set of parameters? are there any other test cases we can use? maybe trying to fail it with edge cases?

also what are we testing? what order do we expect? either rename the test or use pytest.param(....) in the parametrization, and offer a name for each set of parameters describing what we want to achieve. 

e.g:
```
        pytest.param(
            "new_key",
            "new_value",
            id="test fail when new_key doesn't match new_value",
        ),
```